nodebuilder/p2p/metrics.go

```diff
@@ -32,19 +33,27 @@ const (
 	peerIDLabel   = "peer_id"
 )
 
+func enableMetrics(cfg *Config) {
+	cfg.enableMetrics = true
+}
+
 // prometheusMetrics option sets up native libp2p metrics up
 func prometheusMetrics(lifecycle fx.Lifecycle,
 	peerID peer.ID,
 	nodeType node.Type,
 	network Network,
-) error {
+) (prometheus.Registerer, error) {
 	reg := prometheus.NewRegistry()
 	labels := prometheus.Labels{
 		networkLabel:  network.String(),
 		nodeTypeLabel: nodeType.String(),
 		peerIDLabel:   peerID.String(),
 	}
 	wrapped := prometheus.WrapRegistererWith(labels, reg)
+	// Set the default global registerer to the wrapped one with labels. This way all the metrics
+	// registered with the default registerer will be labeled with the provided labels. It is important
+	// because unlike libp2p metrics, bitswap metrics are registered with the default global registerer.
+	prometheus.DefaultRegisterer = wrapped
```

Comment:
Aren't bitswap metrics registered on init causing this overwrite being unaffectual?

Reply:
You are right, I need to ensure order of init:
1. update global
2. inject bitswap Registrer
3. start bitswap